multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -270,6 +284,55 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(commonArea
 	return nil
 }
 
+func (c *StaleResCleanupController) cleanupMemberClusterAnnounce() error {
+	clusterSetList := &mcsv1alpha1.ClusterSetList{}
+	if err := c.List(ctx, clusterSetList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
+	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+		return err
+	}
+
+	for _, memberClusterAnnounce := range memberClusterAnnounceList.Items {
+		var exist bool
+		var needToDelete bool
+		if len(clusterSetList.Items) != 0 {
```

Comment:
I think you can move this check out of loop, and get the members info before this loop.